### DISCUSSION TITLE
Redirect cs.github.com links to github.com for those not signed up

### DISCUSSION THREAD

**Discussion by GMNGeoffrey** [community/community#10415] [Other Feature Feedback, Questions, & Ideas] [answered]
I recently shared a cs.github.com link with a teammate who's not yet in the beta. It redirected them to the general about page for the tech preview (https://cs.github.com/about), which is alright because probably they want to sign up, but it also lost the actual content I was trying to share. How about a way of doing this that doesn't interrupt the direct workflow.

Using the link I was sharing as an example, how about a flow like this. Click https://cs.github.com/google/iree/blob/fe13a3637a2/build_tools/kokoro/gcp_ubuntu/cmake/android/build.sh?q=build_tools%2Fkokoro%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Fbuild.sh#L74 Redirect to something like `https://cs.github.com/about?location=https%3A%2F%2Fgithub.com%2Fgoogle%2Firee%2Fblob%2Ffe13a3637a2%2Fbuild_tools%2Fkokoro%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Fbuild.sh%23L74` or just, `https://cs.github.com/about?location=google%2Firee%2Fblob%2Ffe13a3637a2%2Fbuild_tools%2Fkokoro%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Fbuild.sh%23L74`, since you know the domain. But the landing page should have a *big* link to "continue to GitHub" and an explanation of why they're hitting this page first. The original query from the first link isn't going to do anything, so I think it can be dropped. Maybe the "continue to GitHub" link opens in a new tab, so when the user is done with their flow they can still come back and sign up if they want. It should also have an option to say "always skip this page" so that it doesn't become spam for someone who decides not to sign up or while they're waiting to get off the waitlist.

Related, I notice that if someone isn't logged in they get sent to the login page. I understand that's so you can check if they're in the beta, but it's also annoying because to view code they don't need to log in and it interrupts the flow. Instead, I think it should do something similar. Redirect you to the about page with a further redirect/link to the actual content you were trying to view, as well as message like "already signed up? Log in to continue" And similarly, there should be a pref to always skip this page and go to the content. Obviously it can't be stored in user preferences, so would need to be in local storage or cookies.

I know this stuff is basically only relevant for the beta, but having interop during that time seems important. Otherwise it's GitHub spamming you with not the content you were trying to get to.

**Comment by colinwm** [comment, marked as answer]
Great point about sharing links. 

We have to first log the user in to determine whether they have access to the tech preview. But once they're logged in, if they don't have access to the tech preview, we are now redirecting to the equivalent github.com URL. Hopefully that helps a bit!

**Comment by GMNGeoffrey** [comment]
That's definitely better, thanks! I would suggest also providing a way to bypass login and go to the original link anonymously (and a sticky setting for this in local storage). Someone doesn't generally need to be logged in to view GitHub pages

**Reply by colinwm** [reply]
Unfortunately it's just a limitation of the technology preview, but we'll definitely revisit this when we reach GA!
